### ISSUE TITLE
[List] custom values for LI's ignored

### ISSUE THREAD

**Issue by brendon** [Semantic-Org/Semantic-UI#2802]
In the following code, the bullet value of the second list item will still show as `2`. If we remove the list class, it will show as `3`.

    <ol class="ui list">
      <li>Signing Up</li>
      <li value="3">User Benefits</li>
      <li>User Types
        <ol>
          <li>Admin</li>
          <li>Power User</li>
          <li>Regular User</li>
        </ol>
      </li>
      <li>Deleting Your Account</li>
    </ol>

It seems to be a result of this CSS:

    counter-increment: ordered;
    content: counters(ordered,".") " ";

**Comment by jlukic** [comment]
I had no idea `<li>` could have value. My mind is blown.

**Comment by jlukic** [comment]
I don't believe there's anyway to affect  `:before` and `:after` which are used for bullet stylings without using a `data` attribute. For example see button `or` text. 

Perhaps I could add a `data-value` param for list item that will allow overriding the `content:` param of list

**Comment by brendon** [comment]
Haha! yea, it's a fairly uncommon thing I guess. I need it for showing a list of things extracted from another well known numbered list of things where the original numbers were important but only some of the items need to be included in the current list. Also useful if you want to continue a list in a separate `<ol>`.

A `data-value` attribute would work just fine. As long as it's well documented I can't see the problem. However, could you not access the `value` attribute just as easily as the `data-value` attribute of the element? or am I misunderstanding things? https://developer.mozilla.org/en-US/docs/Web/CSS/attr

**Comment by jlukic** [comment]
Yeah you're right, I guess the difference would be if you're using `<div class="item">` it wouldn't be using an illegal prop. Either way I can just support both equally and call it a day.

**Comment by brendon** [comment]
Yea, that's true. Sounds good. Thanks for following this up so quickly! :D

**Comment by jlukic** [comment]
I will need to work this in with a fix for supporting numbers larger than 100. The before content needs to be auto sized, or else will not work with custom content.

**Comment by brendon** [comment]
Thanks Jack :) I thought of one more caveat, but I can't see how it could be worked around without Javascript. The normal behaviour is for numbering to continue on from the set value. So the next `<li>` would have a number one greater than the previous `<li>`'s set value rather than its current position in the list. The workaround would be to set explicit values from then on.

**Comment by brendon** [comment]
Funnily enough I've come across #2639 in my case also as some list items are double digits. I'll probably just bump up the margin/padding to suit, but I'll keep an eye on your solution too :)

**Comment by brendon** [comment]
I think this can probably be closed as the original issue was fixed. Not sure if you want to create issues for the other things:

- Custom content for list items can blow up margins
- Continue counting from last custom li value (probably outside the scope of Semantic UI)
